### PR TITLE
enable debug assertions in CI

### PR DESCRIPTION
**Pull request by Ekleog-NEAR**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -376,10 +376,11 @@ lto = "fat"
 codegen-units = 1
 
 # A much faster to compile version of `release`.
```
Comment (nagisa) [inline Cargo.toml:378]:
```suggestion
# A much faster to compile version of `release` for development use.
```

Comment (Ekleog-NEAR) [reply Cargo.toml]: Thanks! I had grepped in the docs, but didn’t think of the cargo.toml

**Review by nagisa — APPROVED** [review]
r=me sans the question about that one failing debug assert

**Review by posvyatokum — APPROVED** [review]
Checked that test that I want to fail fails with these changes.
As per [this comment](https://near.zulipchat.com/#narrow/stream/308695-pagoda.2Fprivate/topic/enabling.20debug.20assertions.20in.20CI/near/402795443) I understand, that we can disable that `debug_assert!` as it is:
1. firing in test that it specifically supposed to fire in
2. is part of feature that is going to be disabled with 1.38.0

**Pull request merged**